internal/handler/handler.go

```diff
@@ -164,222 +192,259 @@ func (h *Handler) Process(body []byte, headers map[string]string) (response help
 	var clients *controllers.Client
 	if clients, err = h.githubController.GetGitHubClients(body); err != nil {
 		h.logger.Warn("failed to authenticate", slog.Any("error", err))
-		return helpers.Response{Body: err.Error(), StatusCode: http.StatusInternalServerError}, err
+		return &promotion.Result{
+			Response: helpers.Response{Body: err.Error(), StatusCode: http.StatusUnauthorized},
+		}, promotion.NewInternalError("failed to authenticate. error: %v", err)
 	}
 
 	event, err := github.ParseWebHook(eventType, body)
 	if err != nil {
 		logger.Warn("parsing webhook payload", slog.Any("error", err))
-		return helpers.Response{Body: err.Error(), StatusCode: http.StatusUnprocessableEntity}, fmt.Errorf("invalid payload")
+		return &promotion.Result{
+			Response: helpers.Response{Body: err.Error(), StatusCode: http.StatusUnprocessableEntity},
+		}, promotion.NewInternalError("failed to parse webhook payload. error: %v", err)
 	}
 
 	if err = h.awsController.PutS3Object(eventType, os.Getenv("S3_BUCKET_NAME"), body); err != nil {
-		return helpers.Response{Body: err.Error(), StatusCode: http.StatusInternalServerError}, err
+		logger.Warn("failed to store event in S3", slog.Any("error", err))
+		return &promotion.Result{
+			Response: helpers.Response{Body: err.Error(), StatusCode: http.StatusInternalServerError},
+		}, nil
 	}
 
-	pCtx := &promotion.Context{
-		EventType:  &eventType,
+	result.Context = &promotion.Context{
```

Comment:
Could we not have simplified the refactor here and maintained the cleaner semantic with a simple:
```golang
pCtx := &promotion.Context{
  …
}
result.Context = pCtx
```
…and continued to updated `pCtx` everywhere else?

Reply:
The reason as to why I did will be made much more apparent in the upcoming prefactor PR. :-)